File: 03-sum-nat/sum_nat.lua
-- Lua program to sum numbers from 1 to n
-- Created by M.V.Harish Kumar on 18-06-2023

function SumToN(n)
    local sum = 0
    for i = 1, n do
        sum = sum + i
    end
    return sum
end

io.write("Enter value for `n`: ")
local n = tonumber(io.read("*n"))
local ans = SumToN(n)
print(string.format("The sum from 1 to %d is %d", n, ans))

File: 03-sum-nat/sum_nat.rs
// Rust program to sum numbers from 1 to n
// Created by M.V.Harish Kumar on 19-10-2023
use std::io::{self, Write};

fn sum_to_n(n: u32) -> u32 {
    let mut sum = 0;
    for i in 1..=n {
        sum += i;
    }
    sum
}

fn main() {
    print!("Enter value for `n`: ");
    io::stdout().flush().unwrap();

    let mut n = String::new();
    io::stdin().read_line(&mut n).expect("Failed to read line");

    let n: u32 = n.trim().parse().expect("Invalid Numeric Literal");
    let sum = sum_to_n(n);

    println!("The sum from 1 to {n} is {sum}");
}
